### PR TITLE
Use era-like-memory

### PR DESCRIPTION
**Pull request by StanislavBreadless**
# What ❔

Use memory more similar to the one implemented in zksync-era. Simplifies support of the codebase

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

**Review by EmilLuta — COMMENTED** [review]
The test part looks great to me. Really love the work! I don't have enough context to approve everything else that's been written on `evm` so I'll defer to someone else for the final review.

**Review by joonazan — APPROVED** [review]
LGTM, at least if memory.rs is just copied from era. I did not check that one thoroughly.

**Review by perekopskiy — APPROVED** [review]

**Pull request merged**
